pkg/specgen/generate/kube/volume.go

```diff
@@ -139,6 +142,17 @@ func VolumeFromSecret(secretSource *v1.SecretVolumeSource, secretsManager *secre
 		Source: secretSource.SecretName,
 		Items:  map[string][]byte{},
 	}
+	// Set the defaultMode
+	if secretSource.DefaultMode != nil {
+		if isValidDefaultMode(*secretSource.DefaultMode) {
+			kv.DefaultMode = *secretSource.DefaultMode
+		} else {
+			return nil, fmt.Errorf("invalid DefaultMode for secret %q, must be between 0000 and 0777", secretSource.SecretName)
+		}
+	} else {
+		// Set default DefaultMode if user hasn't given one in the yaml
+		kv.DefaultMode = v1.SecretVolumeSourceDefaultMode
```

Comment:
consider removing this clause and simply define this and then use the conditional above to alter it if needed

Reply:
fixed